RELEASE.md

```diff
@@ -0,0 +1,10 @@
+# Releasing the OpenTelemetry Collector Builder
+
+This project uses [`goreleaser`](https://github.com/goreleaser/goreleaser) to manage the releasing of new versions. To release a new version, simply add a tag named `vX.Y.Z`, like:
```

Comment:
nit: "manage the releasing" -> "manage the release"
Could you put "To release a new..." as a separate paragraph.

Reply:
Done!